spec/offset_manager_spec.rb

```diff
@@ -203,4 +203,29 @@ def partition_offset_info(offset)
       expect(group).to have_received(:commit_offsets).with(expected_offsets)
     end
   end
+
+  describe "#seek_to" do
+    it "seeks to given topic-partition/offset" do
+      topic = "greetings"
+      partition = 0
+      seek_offset = 14
+
+      offset_manager.seek_to(topic, partition, seek_offset)
+
+      offset = offset_manager.next_offset_for(topic, partition)
+
+      expect(offset).to eq seek_offset
+    end
+  end
+
+  describe "#seek_to_default" do
+    it "delegates to seek_to" do
+      topic = "greetings"
+      partition = 0
+
+      expect(offset_manager).to receive(:seek_to).with(topic, partition, -1)
```

Comment:
I'd rather avoid setting call expectations – this is easily testable by calling `next_offset_for`.